internal/upload/upload.go

```
@@ -0,0 +1,622 @@
+
+// openStagedFile opens a .part for positioned writing.
+//
+// The flags are the point: NOT O_APPEND. POSIX sets the file offset to
+// end-of-file before EVERY O_APPEND write, so an explicit Seek is silently
+// ignored. Here that would produce the right bytes by accident — the caller
+// truncates to the durable offset first, making EOF equal that offset — and an
+// accident that happens to be correct is one edit away from not being. With
+// explicit positioning the offset is an assertion: a future change that drops
+// the truncate fails loudly instead of appending to garbage.
+func openStagedFile(path string) (*os.File, error) {
+	return os.OpenFile(path, os.O_WRONLY|os.O_CREATE, 0o600)
```

golangci-lint (2.12.2)

[medium] G304: Potential file inclusion via variable

(gosec)